internal/sequencerapi/api.go

```diff
@@ -0,0 +1,84 @@
+package sequencerapi
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/internal/ethapi"
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+const (
+	txConditionalMaxCost = 1000
+)
+
+var (
+	sendRawTxConditionalCostMeter = metrics.NewRegisteredMeter("sequencer/sendRawTransactionConditional/cost", nil)
+
+	sendRawTxConditionalRequestsCounter = metrics.NewRegisteredCounter("sequencer/sendRawTransactionConditional/requests", nil)
+	sendRawTxConditionalAcceptedCounter = metrics.NewRegisteredCounter("sequencer/sendRawTransactionConditional/accepted", nil)
+)
+
+type sendRawTxCond struct {
+	b ethapi.Backend
+}
+
+func GetSendRawTxConditionalAPI(b ethapi.Backend) rpc.API {
+	return rpc.API{
+		Namespace: "eth",
+		Service:   &sendRawTxCond{b},
+	}
+}
+
+func (s *sendRawTxCond) SendRawTransactionConditional(ctx context.Context, txBytes hexutil.Bytes, cond types.TransactionConditional) (common.Hash, error) {
+	sendRawTxConditionalRequestsCounter.Inc(1)
```

Comment:
Does ERC-4337 or related work not describe error codes to use for certain validation conditions? By wrapping the returned eror with the geth RPC error, error-codes can be specified. At least wrapping it with standard input-errors, to distinguish from server errors, would be helpful to the end user.

Reply:
It does define specific error codes, the old implementation that I worked on defined some of them but not all of them

Reply:
ah nice catch! Didn't realize there were error codes in the api. Went over the spec and added them